File: performanceTest.js
const axios = require('axios');
const { Suite } = require('benchmark');

const baseUrl = 'http://localhost:3000';

const endpoints = ['/parks', '/events', '/spots'];

const suite = new Suite();

endpoints.forEach(endpoint => {
  suite.add(`GET ${endpoint}`, {
    defer: true,
    fn: async (deferred) => {
      try {
        const response = await axios.get(`${baseUrl}${endpoint}`);
        console.log(`Status for ${endpoint}: ${response.status}`);
        deferred.resolve();
      } catch (error) {
        console.error(`Error on ${endpoint}: ${error.message}`);
        deferred.reject(error);
      }
    }
  });
});

suite.on('cycle', (event) => {
  console.log(String(event.target));
}).on('complete', () => {
  console.log('Performance test completed.');
});

suite.run({ async: true });

File: performance_test.py
import requests
import time

# Define the base URL and endpoints to test
base_url = 'http://localhost:3000'
endpoints = ['/parks', '/events', '/spots']

def test_endpoint(endpoint):
    url = f"{base_url}{endpoint}"
    start_time = time.time() 
    try:
        response = requests.get(url)
        elapsed_time = time.time() - start_time 
        print(f"GET {endpoint} - Status Code: {response.status_code}, Response Time: {elapsed_time:.2f} seconds")
    except requests.RequestException as e:
        print(f"Error with {endpoint}: {e}")

for endpoint in endpoints:
    test_endpoint(endpoint)
